### ISSUE TITLE
Bug: Inconsistency of naming when source contains special characters

### ISSUE THREAD

**Issue by DarkVirtue** [kieraneglin/pinchflat#191]
If Pinchflat replaces a special character in the name of a channel, that replacement appears to conflict with the user-configured source name.

For example, [xkcd's What If?](https://www.youtube.com/@xkcd_whatif) has a `?` in the channel name. If I use an output template like this:
```
/shows/{{ source_custom_name }}/Season {{ season_from_date }}/{{ source_custom_name }} - {{ season_episode_from_date }} - {{ title }}.{{ ext }}
```
and set the custom name to `xkcd's What If?`, then I get this directory structure:
```
xkcd's What If#/
├── Season 2023/
│  ├── <snip>
│  └── xkcd's What If# - s2023e1219 - What if NASCAR had no rules？.nfo
└── Season 2024/
   ├── <snip>
   └── xkcd's What If# - s2024e0416 - What if everyone jumped at once？.nfo
xkcd's What If?/
├── banner.jpg
├── fanart.jpg
├── poster.jpg
└── tvshow.nfo
```

I'd expect the software to use the source name for all instances of `source_custom_name` and to keep the season directories inside the same directory as the channel info.

**Comment by kieraneglin** [comment]
Hey there! Thanks for the report (:

Good catch! I'll look into this later today

**Comment by kieraneglin** [comment]
Fixed! This will be in `v0.1.15` which will come out sometime this week.

The issue will be resolved going forward, but it won't retroactively change anything - you'll have to delete and re-add the source to see the change. If you want to fix it now, delete the source and re-add it with a custom name of `xkcd's What If` (no `?`) and that should solve it for now 🤙 

Thanks again for the report!
